tests/unit/test_testing.py

```diff
@@ -104,6 +104,37 @@ def test_defer_job_one_multiple_times(connector):
     assert len(connector.jobs) == 2
 
 
+def test_defer_same_job_with_queueing_lock_second_time_after_first_one_succeeded(
+    connector,
+):
+    job_data = {
+        "task_name": "mytask",
+        "lock": None,
+        "queueing_lock": "some-lock",
+        "args": {},
+        "scheduled_at": None,
+        "queue": "default",
+    }
+
+    # 1. Defer job with queueing-lock
+    job_row = connector.defer_job_one(**job_data)
+    assert len(connector.jobs) == 1
+
+    # 2. Defering a second time should fail, as first one
+    #    still in state `todo`
+    with pytest.raises(exceptions.UniqueViolation):
```

Comment:
I'm unsure whether this is the right `raises` condition here, as the db connector directly raises `AlreadyEnqueued`. Any insights on that?

Reply:
The [aiopg connector](https://github.com/peopledoc/procrastinate/blob/80576fc11f89b50743af11ac8d2c6dce91f3a3b0/procrastinate/aiopg_connector.py#L33) and [psycopg2 connector](https://github.com/peopledoc/procrastinate/blob/80576fc11f89b50743af11ac8d2c6dce91f3a3b0/procrastinate/psycopg2_connector.py#L27) actually raise `UniqueViolation`. So the memory connector should do the same.

`UniqueViolation` is caught by the [manager](https://github.com/peopledoc/procrastinate/blob/3d8d8ad67e555853f5939d70abd43029d435194f/procrastinate/manager.py#L41-L42), which, as a result, is responsible for raising `AlreadyEnqueued`.

Hope this answers your question.

Reply:
I see - yes, it does, thanks